grails-app/controllers/org/pih/warehouse/api/InvoiceApiController.groovy

```diff
@@ -127,8 +128,23 @@ class InvoiceApiController {
         return invoice
     }
 
+    def getAllInvoiceItems() {
+        Location location = Location.get(session.warehouse.id)
+        params.partyFromId = location?.organization?.id
+        List<InvoiceList> invoices = invoiceService.getInvoices(params)
```

Comment:
I spoke about that with @drodzewicz, but I want to make my point here, so that everyone can give their opinion - in my opinion it should be dealt directly via `InvoiceItem.createCriteria()` and 
```groovy
invoice {
  'in'("id" ,params.invoicesIds)
}
```
clause.
Plus I would monitor the performance here, because we might run into many N+1 - if we do, we could play with the data services and joins, but for now, I'd like to get others opinion about the approach of first getting the invoices and then getting the invoice items.

Reply:
I don't think we are at risk of N+1 queries, if I am not missing anything this should be max 2 queries one for invoices and one for invoiceItems.

My goal was to reuse the query filter logic on  `invoiceService.getInvoices(params)` since we are expecting to get only those InvoiceItems that are a part of filtered invoices from invoice/list page, basically aggregate these queries.
First get invoices (filtered if necessary) and then based on these invoices get all of the invoiceItems

Reply:
@drodzewicz On the other hand if this is for "import all" case you could also just do: `get invoices` -> `return invoices.invoiceItems`. Is the pagination required here? However if this is going to land in its own api controller, or the pagination params are required, then: 

I agree with @kchelstowski on that one in the long term. If you need to do some further invoice filtering you still can do it in the suggested by Kacper way as a:
```groovy
invoice {
  'in'('id', params.invoiceIds) // if there are invoice ids passed
  whatever other invoice filter you have there
}
```
Instead of doing `get invoices` -> ` get invoice items` by these invoices.